fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java

```diff
@@ -214,11 +219,58 @@ protected Response getContent(final String rangeValue,
         return ok(outputStream).build();
     }
 
+
+    protected URI getURLAccessType(final MediaType mediaType) {
+        return URI.create(getAccessTypeValue(mediaType.getParameters(), URL_ACCESS_TYPE));
+    }
+
+    /**
+     * Throws a NotSupportedException if media type is an invalid message/external-body type. To be valid it must
+     * contain the access-type parameter that equals a valid access type and a URL parameter that contains a URL.
+     * 
+     * @param mediaType The media type to be validated
+     * @throws NotSupportedException
+     */
+    protected void checkExternalBody(final MediaType mediaType) {
+        if (isExternalBody(mediaType)) {
+            final Map<String, String> params = mediaType.getParameters();
+            final String accessType = params.get(ACCESS_TYPE);
+            if (accessType == null) {
+                throw new BadRequestException("You must specify the access-type param when using " +
+                        MESSAGE_EXTERNAL_BODY + " content type:" +
+                        mediaType.toString());
+            }
+
+            if (!accessType.equalsIgnoreCase(URL_ACCESS_TYPE)) {
+                throw new NotSupportedException("Unsupported media type:  the " + accessType +
+                        " access type  is not supported: " + mediaType.toString());
+            }
+
+            if (getAccessTypeValue(params, accessType) == null) {
```

Comment:
Oh my bad, here is a call. 

So you could replace this check with the stream above but instead of `orElse(null);` use
`.orElseThrow(() -> new BadRequestException("When specifying the  " + accessType +
                        " access type you must include a " + URL_ACCESS_TYPE + " key/value pair: " + mediaType
                        .toString()));`


I dunno, I'm told streams are better than loops.

Reply:
Actually isn't this check the same as the one on line 238?